test_conformance/run_conformance.py

```diff
@@ -1,302 +1,341 @@
+            match_ = re.search(".*FAILED.*", line)
+            if match_:
+                failures_this_run = failures_this_run + 1
+            match_ = re.search(".*(PASSED).*", line)
+            if match_:
+                if not output_this_run:
+                    print("")
+                    output_this_run = True
+                print("               " + line.replace('\n', ''))
+            # Write it to the log
+            log_file_.write("     " + line + "\n")
+            log_file_.flush()
+            line = ""
+            pointer = pointer + 1
+        # If we are at the end of the file, then re-open it to get new data
+        elif char_read == "":
+            more_to_read = False
+            read_output.close()
+            time.sleep(1)
+            try:
+                os.fsync(output_fd)
+                read_output = open(output_name, 'r')
+                # See if there is more to read. This happens if the process ends and we have data left.
+                read_output.seek(pointer)
+                if read_output.read(1) != "":
+                    more_to_read = True
+            except IOError:
+                write_screen_log("\n           ==> ERROR: could not reopen output file from test.")
+                return -1
+        else:
+            line = line + char_read
+            pointer = pointer + 1
+    # Now we are done, so write out any remaining data in the file:
+    # This should only happen if the process exited with an error.
+    os.fsync(output_fd)
+    while read_output.read(1) != "":
+        log_file_.write(read_output.read(1))
+    # Return the total number of failures
+    if p.returncode == 0 and failures_this_run > 0:
+        write_screen_log("\n           ==> ERROR: Test returned 0, but number of FAILED lines reported is " + str(
+            failures_this_run) + ".")
+        return failures_this_run
+    return p.returncode
+
+
+def run_tests(tests_):
+    global curent_directory
+    global process_pid
+    # Run the tests
+    failures_ = 0
+    previous_test = None
+    test_number = 1
+    for t in tests_:
+        # Print the name of the test we're running and the time
+        (test_name_, test_dir_) = t
+        if test_dir_ != previous_test:
+            print("==========   " + test_dir_)
+            log_file.write("========================================================================================\n")
+            log_file.write("========================================================================================\n")
+            log_file.write("(" + get_time() + ")     Running Tests: " + test_dir_ + "\n")
+            log_file.write("========================================================================================\n")
+            log_file.write("========================================================================================\n")
+            previous_test = test_dir_
+        print("(" + get_time() + ")     BEGIN  " + test_name_.ljust(40) + ": "),
```

Comment:
I notice there's a trailing `,` here which won't behave in Python 3 (or in Python 2 using the print function from futures) in the way that is intended.

I have a fix for this in PR #1492, which I would like for us to apply first before we finish this review.  Please ensure the fix isn't undone in this change after we have closed PR #1492.